src/Entity/DatasetSubmission.php

```diff
@@ -2764,14 +2754,38 @@ public function getDistributionPoints()
         return $this->distributionPoints;
     }
 
+     /**
+     * Getter for the (first) erddap url link.
+     *
+     * @return DatasetLink|null
+     */
+    private function getErdappUrlLink() : ?DatasetLink
+    {
+        $datasetLinks = $this->getDatasetLinks()->filter(function (DatasetLink $datasetLink) {
+            return $datasetLink->getName() == "ERDDAP";
+        });
+
+        if ($datasetLinks->count() > 0) {
+            return $datasetLinks->first();
+        }
+
+        return null;
+    }
+
     /**
      * Getter for the erddap url.
      *
      * @return string|null
      */
     public function getErddapUrl(): ?string
```

Comment:
Are we using this getter anywhere?

Reply:
No, not currently, but can be used later on the data landing page or, or whoever wants the link. I should probably make getErddapDatasetLink public, incase someone wants to use that.

Reply:
Okay, makes sense. I think all the ERDDAP stuff should be moved to DatasetLinks entity

Reply:
So these are actually used in the ISO metadata generator.
Also in Dataland, but the button is not visible.